dev/storage.h

```diff
@@ -868,7 +952,11 @@ namespace sqlite_orm {
                         }
                     }
                     if(gottaCreateTable) {
-                        res = decltype(res)::dropped_and_recreated;
+                        if(preserve) {
+                            res = decltype(res)::dropped_and_recreated;
+                        } else {
+                            res = decltype(res)::table_data_loss;
```

Comment:
as I said `table_data_loss` is returned when `dropped_and_recreated` returns in `dev` with `preserve = false`. Also `table_data_loss` sounds like a data state not schema sync result so I don't understand why you added it to `sync_schema_result`. If user want to know whether data was lost he/she can do it right now in `dev` but checking or `dropped_and_recreated` and `preserve == false`

Reply:
nit picking, the semantics of table_data_loss is a schema sync result just like dropped_and_recreated. It triggers a different treatment... see backup_table() and drop_create_with_loss() ... the latter doesn't backup and restore only drops/recreates if FK allows (rememeber FK is on). 
Actually because FK is on, then dropped and recreated only works when no  FK constraints are violated

Reply:
I see the difference but table_data_loss has nothing common with schema. Schema is a list of objects of four types: tables, views, indexes and triggers. Please note that schema doesn't contain rows. No rows = no data. Every existing enum value of `sync_schema_result` a unique operation performed with schema. If you compare all enum value with each other you'll notice that there are no two equal enum values. But if you compare `table_data_loss` with `dropped_and_recreated` you'll find that `table_data_loss` is actually `dropped_and_recreated` in regards of schema. This is why `table_data_loss` is not a sync schema result. So it has nothing to do in `sync_schema_result` declaration. If someone need to know whether data (that is, rows) was dropped also he/she can check if result is `dropped_and_recreated` and `preserve` is `false`.

Reply:
if you think that I am nit picking you can make your custom fork where you can do whatever you want. Currently `sqlite_orm` has 200+ forks where people can commit what they want without `nit picking`. Actually you are the first who said that. Sometimes when you think that the world is wrong and you are right probably everything is reversed: the world is ok but you do something wrong

Reply:
what if I eliminate that value from the enum? Will you be finally at peace with it?

Reply:
if you eliminate `table_data_loss` then the issue with `table_data_loss` will stop existing

Reply:
table_data_loss was removed! FK toggling option too!